drivers/adc/ad4858/ad4858.c

```diff
@@ -383,6 +383,67 @@ int ad4858_enable_test_pattern(struct ad4858_dev *dev, bool test_pattern)
 
 	return 0;
 }
+/**
+ * @brief Enable/Disable channel sleep.
+ * @param dev - Pointer to the device structure.
+ * @param chn - Input channel.
+ * @param sleep_status - Sleep status.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4858_enable_ch_sleep(struct ad4858_dev* dev, uint8_t chn,
+			   enum ad4858_ch_sleep_value sleep_status)
+{
+	int ret;
+	uint32_t mask;
+	uint32_t val;
+	uint32_t data;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad4858_reg_read(dev, AD4858_REG_CH_SLEEP, &data);
+	if (ret)
+		return ret;
+
+	dev->chn_sleep_value[chn] = sleep_status;
```

Comment:
Check if `chn` is not out of the array bounds. There are other places where this is the case